pkg/deviceplugin/api.go

```diff
@@ -102,6 +102,19 @@ func (tree DeviceTree) AddDevice(devType, id string, info DeviceInfo) {
 		tree[devType] = make(map[string]DeviceInfo)
 	}
 
+	if info.cdiSpec != nil {
+		devLength := len(info.cdiSpec.Devices)
+		if devLength == 0 {
+			klog.Warning("No CDI devices defined in spec, removing spec")
+
+			info.cdiSpec = nil
+		} else if devLength > 1 {
+			klog.Warning("Including more than one CDI device per spec is not supported")
```

Comment:
It's possible but it would mean changing the function signature. Giving more than one device per spec could be considered a fatal (development) error and maybe should break the execution.